src/useStorage/storage.ts

```diff
@@ -0,0 +1,92 @@
+import { globalExists } from '../utilities/global'
+
+export type StorageType = 'session' | 'local'
+
+export class StorageManager {
```

Comment:
this is 100% cool as is, just sharing what my approach would have been in case you think it's groovy. 

what if instead of defining a single class that changes functionality of each method based on a single property `type`, you created a separate class for localStorage and sessionStorage that both implemented `StorageManager` or something like that. StorageManager could have protected methods to abstract away the JSON work so each implementation doesn't have to duplicate that logic.

Reply:
Both localStorage and sessionStorage have the exact same api. So the two classes would be identical except for calling `localStorage` or `sessionStorage`. Considering that's the only difference between the two storage types. Once class is favorable over three IMO.